### ISSUE TITLE
ErnieForGeneration模型如何导出静态图或者ONNX？

### ISSUE THREAD

**Issue by shukhg** [PaddlePaddle/PaddleNLP#2697]
（1）环境
paddlepaddle-gpu=2.2.2.post112
paddlenlp=2.3.3
paddle2onnx=0.9.8
cuda=11.2

（2）问题
ERNIE-GEN 模型输入维度不固定，无法导出静态图。
主要是因为模型的输入数目是不固定的（例如下面的例子 model() 中参数不固定 ）
在这种情况了无法导出静态图、也无法导出 ONNX，请问这种该如何处理？

参考链接：
https://aistudio.baidu.com/aistudio/projectdetail/1339888

链接中相关代码（模型输入参数不固定）：
    _, logits, info = model(q_ids, q_sids)
    d_batch, d_seqlen = q_ids.shape
    seqlen = paddle.sum(paddle.cast(q_ids != 0, 'int64'), 1, keepdim=True)
    has_stopped = np.zeros([d_batch], dtype=np.bool)
    gen_seq_len = np.zeros([d_batch], dtype=np.int64)
    output_ids = []

    past_cache = info['caches']

    cls_ids = paddle.ones([d_batch], dtype='int64') * sos_id
    attn_ids = paddle.ones([d_batch], dtype='int64') * attn_id
    ids = paddle.stack([cls_ids, attn_ids], -1)
    for step in range(max_decode_len):
        bias = gen_bias(q_ids, ids, step)
        pos_ids = paddle.to_tensor(
            np.tile(
                np.array(
                    [[step, step + 1]], dtype=np.int64), [d_batch, 1]))
        pos_ids += seqlen
        _, logits, info = model(
            ids,
            paddle.ones_like(ids) * tgt_type_id,
            pos_ids=pos_ids,
            attn_bias=bias,
            past_cache=past_cache)

**Comment by wawltor** [comment]
内部讨论了一下，目前ERNIE-Gen在短期内是很难支持导出静态图或者Onnx，是因为ENRIE-GEN本身模型用了相对相对特殊的解码方式，导致很难导出静态图的模型；
目前PaddleNLP在docoder预测加速主要是集中在FasterGeneration的工作，具体可以看看  https://github.com/PaddlePaddle/PaddleNLP/tree/develop/faster_generation , 这里的unimo-text做文本生成可以看看
